src/server.jl

```diff
@@ -436,25 +443,23 @@ function serve(fw::FileWatcher=SimpleWatcher(file_changed_callback);
     start(fw)
 
     # make request handler
-    req_handler = HTTP.RequestHandlerFunction() do req
+    req_handler = streamhandler() do req
         req = preprocess_request(req)
         serve_file(fw, req; inject_browser_reload_script = inject_browser_reload_script, allow_cors = allow_cors)
     end
 
     server = Sockets.listen(parse(IPAddr, host), port)
     url = "http://$(host == string(Sockets.localhost) ? "localhost" : host):$port"
     println("✓ LiveServer listening on $url/ ...\n  (use CTRL+C to shut down)")
-    @async HTTP.listen(host, port;
-                       server=server, readtimeout=0, reuse_limit=0) do http::HTTP.Stream
-        # reuse_limit=0 ensures that there won't be an error if killing and restarting the server.
```

Comment:
And what about this?

Reply:
I think as long as `close(server)` is called vs `HTTP.forceclose(server)`, we should be ok now. `close(server)` initiates a "graceful" shutdown where active connections are _marked_ to close and close _themselves_ as they get the chance in their own event loop, so that should avoid the "hard" errors we may have seen previously by connections that didn't have a reuse_limit.